### PR TITLE
[UR] Pass pi_mem_properties in pi2ur for piMemBufferCreate

### PR DESCRIPTION
**Pull request by nrspruit**
- Remove the restriction in pi2ur such that piMemBufferCreate can pass the pi_mem_properties as the pNext properties to the UR Adapters.

### CODE REVIEW

**Comment by nrspruit** [comment]
@intel/unified-runtime-reviewers , please let me know if you see any issues with this change that could impact the adapters. This re-enables the usage of the pi mem properties used in the opencl adapter at least.

File: sycl/plugins/unified_runtime/pi2ur.hpp [hunk 1]
```diff
@@ -2697,6 +2693,43 @@ inline pi_result piMemBufferCreate(pi_context Context, pi_mem_flags Flags,
...
   UrProps.pHost = HostPtr;
+
+  ur_buffer_channel_properties_t bufferChannelProperties{};
+  bufferChannelProperties.stype = UR_STRUCTURE_TYPE_BUFFER_CHANNEL_PROPERTIES;
+  ur_buffer_alloc_location_properties_t bufferLocationProperties{};
+  bufferLocationProperties.stype =
+      UR_STRUCTURE_TYPE_BUFFER_ALLOC_LOCATION_PROPERTIES;
+  if (properties != nullptr) {
+    constexpr uint64_t defaultBufferLocation = 7777;
+    constexpr uint32_t defaultBufferMemChannel = 7777;
```
Comment (aarongreig) [inline sycl/plugins/unified_runtime/pi2ur.hpp:2704]: I understand that this is astronomically unlikely to cause a problem by being a legit value for these properties one day, but I still think it's asking for trouble. I'd rather these two were boolean flags that get set if we find the corresponding property

Comment (nrspruit) [reply sycl/plugins/unified_runtime/pi2ur.hpp]: Thanks @aarongreig ,

I have updated accordingly.

**Review by aarongreig — APPROVED** [review]

**Comment by nrspruit** [comment]
Test failure for gen12 is an env issue:
# .---command stdout------------
# | Abort was called at 253 line in file:
# | ../../neo/level_zero/core/source/builtin/builtin_functions_lib_impl.cpp
# `-----------------------------

that failure is in the level zero gpu driver and does not have anything to do with the changes in this patch.

**Pull request merged**